### PR TITLE
Add the service account caller explicitly instead of its token

### PR DESCRIPTION
**Pull request by umbreak**

### CODE REVIEW

File: delta/src/main/resources/app.conf [hunk 1]
```diff
@@ -40,11 +40,12 @@ app {
     default-timeout = 30 seconds
   }
 
-  # Service account settings
-  service-account {
-    # the service account token, if needed (when not required, anonymous subject will be used)
-    token = null
-  }
+  # caller settings for when the service wants perform service calls. Do not asign a value if you want the caller to be anonymous
```
Comment (imsdu) [inline delta/src/main/resources/app.conf:43]: To delete or to uncomment I guess

File: delta/src/main/scala/ch/epfl/bluebrain/nexus/delta/Main.scala [hunk 2]
```diff
@@ -169,7 +166,8 @@ object Main {
         ArchiveCache[Task].runSyncUnsafe()(Scheduler.global, pm)
       )
     implicit val pc: ProjectCache[Task]           = cache.project
-    implicit val projectResolution                = ProjectResolution.task(repo, cache.resolver, cache.project, acls, saCaller)
+    implicit val projectResolution                =
```
Comment (imsdu) [inline delta/src/main/scala/ch/epfl/bluebrain/nexus/delta/Main.scala:169]: Formatting is weird here

File: delta/src/main/resources/app.conf (hunk 1 above)
Comment (umbreak) [reply delta/src/main/resources/app.conf:43]: No. It is left as a documented example of which parameters needs to be set when defining a caller

File: delta/src/main/scala/ch/epfl/bluebrain/nexus/delta/Main.scala (hunk 2 above)
Comment (umbreak) [reply delta/src/main/scala/ch/epfl/bluebrain/nexus/delta/Main.scala:169]: Yeah. It is scalafmt formatting. Not a big deal

**Pull request merged**